src/pepr/operator/crd/validators/package-validator.ts

```diff
@@ -115,5 +115,20 @@ export async function validator(req: PeprValidateRequest<UDSPackage>) {
     }
   }
 
+  const monitors = pkg.spec?.monitor ?? [];
+
+  // Ensure the descriptions are generating unique names
+  const monitorDescriptions = new Set<string>();
+
+  for (const monitor of monitors) {
+    const description = sanitizeResourceName(monitor.description || "null");
```

Comment:
Good question. I think it's worth evaluating again - when we implemented this initially we intended for it to be a way to name resources more descriptively rather than getting an autogenerated name. I think we would want that functionality regardless, but maybe `name` would've been more appropriate, with a description occupying an annotation instead. We do trim/sanitize regardless so a long sentence would fit into the resource name, but it might end up looking weird if description is set to something like `This service monitor is used to monitor the application metrics for foobar app`.

I would say we keep the current naming logic in this PR but open to a follow-on ticket/PR to evaluate a different precedence/naming convention.